src/components/v2v3/V2V3Project/ProjectDashboard/hooks/usePayProjectModal/usePayProjectTx.ts

```diff
@@ -97,15 +110,54 @@ export const usePayProjectTx = ({
         .map(({ id, quantity }) => Array<number>(quantity).fill(id))
         .flat()
 
-      const delegateMetadata = encodeJb721DelegateMetadata(
-        {
-          tierIdsToMint,
-          allowOverspending:
-            nftRewards.length > 0 ? DEFAULT_ALLOW_OVERSPENDING : undefined,
-        },
-        JB721DelegateVersion,
+      // Total tokens that should be minted by pay() tx, including reserved tokens
+      const requiredTokens = fundingCycleMetadata
+        ? parseWad(receivedTickets)
+            .mul(MAX_RESERVED_RATE)
+            .div(fundingCycleMetadata.reservedRate)
+        : undefined
+
+      const expectedTokensFromSwap = weiAmount.div(
+        BigNumber.from(priceQuery?.projectTokenPrice.numerator.toString()).div(
+          BigNumber.from(priceQuery?.projectTokenPrice.denominator.toString()),
+        ),
       )
 
+      const shouldUseBuybackDelegate =
+        projectId &&
+        priceQuery &&
+        requiredTokens &&
+        BUYBACK_DELEGATE_ENABLED_PROJECT_IDS.includes(projectId) &&
```

Comment:
(non-blocking) instead of this, I think we could use the Delegates Registry to check if their fc datasource is a buy-back delegate. But this works for now 🙏